### PR TITLE
scala: Add accumulate mentoring notes

### PR DESCRIPTION
**Pull request by ricemery**

### CODE REVIEW

File: tracks/scala/exercises/accumulate/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    loop(list, ListBuffer.empty).toList
+  }
+
+}
+```
+
+### Common suggestions
+
+- Appending to a list in Scala is linear in proportion to the collection size. Prepend is constant time. Suggest prepending to build the list in reverse order. Then reverse the resulting list before returning. Alternately ListBuffer could be used.  https://docs.scala-lang.org/overviews/collections/performance-characteristics.html
```
Comment (SleeplessByte) [inline tracks/scala/exercises/accumulate/mentoring.md]: Make this link clickable and consider shortning the text
```markdown
[Reference: performance characteristics](https://docs.scala-lang.org/overviews/collections/performance-characteristics.html)
```

File: tracks/scala/exercises/accumulate/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  @tailrec
+  private def accumulateHelper[A, B](f: (A) => B, list : List[A], result : List[B]): List[B] =
+    list match {
+      case Nil => result.reverse
+      case x :: xs => accumulateHelper(f, xs, f(x) :: result)
+    }
+}
+```
+
+#### Solution using mutable collection
```
Comment (ErikSchierboom) [inline tracks/scala/exercises/accumulate/mentoring.md]: This is the same header as the previous solution. Perhaps use another header?

File: tracks/scala/exercises/accumulate/mentoring.md (hunk 1 above)
Comment (ErikSchierboom) [inline tracks/scala/exercises/accumulate/mentoring.md]: Perhaps a link to tail-recursiveness in Scala could be added here? If there is a nice research available.

Comment (SleeplessByte) [reply tracks/scala/exercises/accumulate/mentoring.md]: https://alvinalexander.com/scala/fp-book/tail-recursive-algorithms perhaps?

File: tracks/scala/exercises/accumulate/mentoring.md (hunk 2 above)
Comment (ricemery) [reply tracks/scala/exercises/accumulate/mentoring.md]: I am confused. The first sample solution states 'immutable' collection. The second solution states ' mutable' collection? I don't understand what you are pointing out.

Comment (ErikSchierboom) [reply tracks/scala/exercises/accumulate/mentoring.md]: I completely misread that, sorry about that! I read immutable twice.

File: tracks/scala/exercises/accumulate/mentoring.md (hunk 1 above)
Comment (ricemery) [reply tracks/scala/exercises/accumulate/mentoring.md]: Thanks. Will do.

Comment (ricemery) [reply tracks/scala/exercises/accumulate/mentoring.md]: Thanks. I am going to add this.

**Comment by ricemery** [comment]
No worries. Thanks for reviewing!

File: tracks/scala/exercises/accumulate/mentoring.md (hunk 2 above)
Comment (ricemery) [reply tracks/scala/exercises/accumulate/mentoring.md]: No worries. Thanks for reviewing.

**Comment by ErikSchierboom** [comment]
I'll wait for @SleeplessByte's approval, as he had some comments.

**Comment by SleeplessByte** [comment]
Perfect! Thanks @ricemery (and @ErikSchierboom) ❤️

**Comment by ricemery** [comment]
Thanks @ErikSchierboom  and @SleeplessByte  !!

**Pull request merged**